WalletLibrary/WalletLibraryTests/Requests/Handlers/OpenId4VCIHandlerTests.swift

```diff
@@ -57,21 +57,198 @@ class OpenId4VCIHandlerTests: XCTestCase
+        let configuration = LibraryConfiguration(networking: mockNetworking)
+        let handler = OpenId4VCIHandler(configuration: configuration)
+        
+        do
+        {
+            // Act
+            let _ = try await handler.handle(rawRequest: rawRequest)
+        }
+        catch
+        {
+            // Assert
+            XCTAssert(error is OpenId4VCIValidationError)
+            let validationError = error as! OpenId4VCIValidationError
+            XCTAssertEqual(validationError.code, "credential_metadata_malformed")
+            XCTAssertEqual(validationError.message, "Request does not contain expected credential configuration.")
+        }
+    }
+    
+    func testHandle_WithMismatchedAuthServers_ThrowsError() async throws
+    {
+        // Arrange
+        let rawRequest = createJSONCredentialOffer()
+        let metadata = createCredentialMetadata(authorizationServer: "mismatchedAuthServer")
+        let expectedResult = (metadata, CredentialMetadataFetchOperation.self)
+        let mockNetworking = MockLibraryNetworking.create(expectedResults: [expectedResult])
+        let configuration = LibraryConfiguration(networking: mockNetworking)
+        let handler = OpenId4VCIHandler(configuration: configuration)
+        
+        do
+        {
+            // Act
+            let _ = try await handler.handle(rawRequest: rawRequest)
+        }
+        catch
+        {
+            // Assert
+            XCTAssert(error is OpenId4VCIValidationError)
+            let validationError = error as! OpenId4VCIValidationError
+            XCTAssertEqual(validationError.code, "credential_metadata_malformed")
+            XCTAssertEqual(validationError.message, "Authorization servers in Credential Metadata does not contain expectedAuthorizationServer")
+        }
+    }
+    
+    func testHandle_WithSignedMetadataProcessorError_ThrowsError() async throws
+    {
+        // Arrange
+        let rawRequest = createJSONCredentialOffer()
+        let metadata = createCredentialMetadata()
+        
+        let expectedResult = (metadata, CredentialMetadataFetchOperation.self)
+        let mockNetworking = MockLibraryNetworking.create(expectedResults: [expectedResult])
+        let configuration = LibraryConfiguration(networking: mockNetworking)
+        
+        let processor = MockSignedMetadataProcessor(shouldThrow: true)
+        
+        let handler = OpenId4VCIHandler(configuration: configuration,
+                                        signedMetadataProcessor: processor)
+        
+        do
+        {
+            // Act
+            let _ = try await handler.handle(rawRequest: rawRequest)
+        }
+        catch
+        {
+            // Assert
+            XCTAssert(error is MockSignedMetadataProcessor.MockError)
+            XCTAssertEqual(error as? MockSignedMetadataProcessor.MockError, .ErrorExpected)
+        }
+    }
+    
+    func testHandle_WithNoScopeValue_ReturnsVerifiedId() async throws
```

Comment:
maybe you have already added. Could you please add a test for missing any required field in all new data models if not added already?

Reply:
Should have tests for all of these.